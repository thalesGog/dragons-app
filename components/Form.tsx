import styled from 'styled-components'

const Form = styled.form`
  display: flex;
  flex-direction: column;
  max-width: 600px;
  margin: 0 auto;
  justify-content: center;
  width: 100%;
  height: 100vh;
  a,
  label,
  input,
  button {
    padding: 6px 0 6px 0;
  }
  a {
    color: ${({theme}) => theme.colors.primary};
    text-decoration: none;
    font-weight: 600;
  }
  button {
    margin-top: 18px;
    margin-bottom: 24px;
  }
  input {
    margin-top: 8px;
    font-size: 1rem;
    text-indent: 0.8rem;
    padding: 0.8rem 0;
    outline: none;
    width: 100%;
    border: 1px solid #dddce0;
    border-radius: 2px;
    transition: border-color 0.15s;
    font-weight: 400;
    box-sizing: border-box;
    height: 50px;
  }
  label {
    display: flex;
    flex-direction: column;
    text-align: left;
    font-weight: 600;
  }
  img {
    max-width: 70px;
    margin: 0 auto;
  }
`

export default Form
