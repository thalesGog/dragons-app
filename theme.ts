import {DefaultTheme} from 'styled-components'

const theme: DefaultTheme = {
  fontFamily: 'Roboto, sans-serif',
  backgroundColor: '#fff0e5',
  colors: {
    primary: '#10162f',
    secondary: '#3a10e5',
  },
}

export default theme
