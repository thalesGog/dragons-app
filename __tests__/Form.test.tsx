import React from 'react'
import {ThemeProvider} from 'styled-components'
import theme from '../theme'
import {render} from '@testing-library/react'
import '@testing-library/jest-dom/extend-expect'
import 'jest-styled-components'
import Form from '../components/Form'

test('Render Form component', async () => {
  const {container} = render(
    <ThemeProvider theme={theme}>
      <Form />
    </ThemeProvider>,
  )
  expect(container).toMatchSnapshot()
})
