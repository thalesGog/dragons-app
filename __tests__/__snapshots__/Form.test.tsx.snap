// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Render Form component 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  max-width: 600px;
  margin: 0 auto;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  width: 100%;
  height: 100%;
  -webkit-transform: translateY(40%);
  -ms-transform: translateY(40%);
  transform: translateY(40%);
}

.c0 a,
.c0 label,
.c0 input,
.c0 button {
  padding: 6px 0 6px 0;
}

.c0 a {
  color: #10162f;
  -webkit-text-decoration: none;
  text-decoration: none;
  font-weight: 600;
}

.c0 a:hover {
  color: #3a10e5;
}

.c0 button {
  margin-top: 18px;
  margin-bottom: 24px;
}

.c0 input {
  margin-top: 8px;
  font-size: 1rem;
  text-indent: 0.8rem;
  padding: 0.8rem 0;
  outline: none;
  width: 100%;
  border: 1px solid #dddce0;
  border-radius: 2px;
  -webkit-transition: border-color 0.15s;
  transition: border-color 0.15s;
  font-weight: 400;
  box-sizing: border-box;
  height: 50px;
}

.c0 label {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  text-align: left;
  font-weight: 600;
}

.c0 img {
  max-width: 70px;
  margin: 0 auto;
}

<div>
  <form
    class="c0"
  />
</div>
`;
